pkg_defs/ska3-core/combine_arch.py

```diff
@@ -41,6 +42,29 @@ def main():
         with open(filename) as fh:
             environments[platform] = {p['name']: p for p in json.load(fh)}
 
+    subtract_environments = {}
```

Comment:
Every time you find yourself copy/pasting a section of code and changing a couple of variables, ask yourself if it can be refactored. I believe the answer here is yes.

Reply:
Sure.  I think my line for looking twice is a little different from yours on a bad utility script but I can make this nicer.

Reply:
Refactored.